specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json

```diff
@@ -443,6 +443,48 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataFactory/factories/{factoryName}/upgrade": {
+      "post": {
+        "tags": [
+          "factories"
+        ],
+        "operationId": "Factories_Upgrade",
+        "x-ms-examples": {
+          "Factories_Upgrade": {
+            "$ref": "./examples/Factories_Upgrade.json"
+          }
+        },
+        "description": "Upgrade preview version factory to G.A. version.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/factoryName"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/Factory"
+            }
```

Comment:
I don't think you would like to return the resource representation here.. just 200 (OK) should be good.

Reply:
+ @ravbhatnagar

Reply:
Can you add more details on the upgrade behavior ? whether it is a one time thing, in the life cycle of a resource ? 
POST can't be expressed in ARM template and is discouraged for things that customer would like to express them in declarative manner (in template). 

---
In reply to: [264448182](https://github.com/Azure/azure-rest-api-specs/pull/5297#discussion_r264448182) [](ancestors = 264448182)